### ISSUE TITLE
[Enhancement request] Wallpaper

### ISSUE THREAD

**Issue by OpenMindNL** [athombv/homey#1810]
**What is your Homey version (Settings → System)?**

1.5.1

**What did you try to achieve?**

Import one user wallpaper

**What was the result?**

No way found to use another wallpaper than those that are included.

**What did you expect as result?**

Hoped for at least one user selection.

**Any other remarks**

It would be very nice if a user could import a personal wallpaper.
One selection would be enough... other wallpapers could be selected by replacing the personal wallpaper.

**Please attach a Homey System log (Settings -> System -> Send Log)**

No wallpapers in log :grin:

**Comment by WeeJeWel** [comment]
However nice, we won't add this functionality, because it will mean proxying large amounts of data through the Athom Cloud services. Current wallpapers are downloaded from a CDN.

**Comment by OpenMindNL** [comment]
In that case: Please provide a deep black one. The one that I have currently selected is not because I like it, but because it's the darkest one.

**Comment by OpenMindNL** [comment]
I've been thinking a little bit further about the wallpaper options.

1. How about optimizing the current selections with a brightness setting?
- The user selects a desired wallpaper (just like now).
- The user sets the brightness level, that could range from 25% to 100% (normal)
- Homey calculates the final wallpaper by applying: rgbToHsv(pixel), adjust V, hsvToRgb(pixel).
- Homey calculates if it should use black or white text color.

2. A second thought for a simple but even greater enhancement:
Replace the monochrome wallpapers (like oceandeep) by their greyscale version and let the user colorize them by converting the grey-shades to a selected color-tone AND adjust brightness.

Both of these options would need storage for one extra wallpaper... the calculated result. Maybe it would even use LESS storage because some wallpapers look very much alike, except for their color and brightness... a few excisting wallpapers could be removed....
Both brightness and colorizing calculations are very easy and fast to do. Takes a few seconds extra during wallpaper selection only.

Pleeeeaaase 😀

**Comment by OpenMindNL** [comment]
Just popped into my mind... The wallpaper is used as background in HTML... So why not let the client do the color/brightness setting? Even more resource friendly to Homey. Use a canvas for background... 😮 

and one step further would be to let the user select a local background 😁

**Comment by bvdbos** [comment]
Did you find any code where a background jpg or png is created on the fly?

**Comment by OpenMindNL** [comment]
Creating images is easy with HTML5 / Javascript. It's all in CANVAS.
I've used this technique a few years ago when I created and helped creating a lot of smartwatch 'clockworks' for Samsung Gear, Gear2 and Gear S (the square / rectangular watches with Tizen OS).

All the graphics were created on the fly and copied into image-objects (*.png). It was implemented so users could change the design/colors on the fly.

**Comment by WeeJeWel** [comment]
This is not going to happen, as we're moving to the Homey Smartphone App only :-)

You're always free to create a custom stylesheet using Stylish or similar plugins.

**Comment by OpenMindNL** [comment]
I just don't understand you guys...
Waste resources to a project that has nothing to do with home automation (Homey Music), but not willing to implement useful things that people use all the time (wallpaper in app and homey desktop)

And moving to smartphone app only.... from this I understand that I wasted many, many, many hours to make LEDitor usefull for everybody instead of just for me.... 😞

**Comment by WeeJeWel** [comment]
@OpenMindNL Actually LEDitor is a perfect example for our new architecture. The idea is that you will host a web-portal, e.g. leditor.com, that makes use of the Web API (https://developer.athom.com) so developers have access to a full-desktop experience while developing. Similar things are probably going to happen to other projects like HomeyDash, and we already did it with HomeyScript.

**Comment by OpenMindNL** [comment]
Define tools?
As I see it, LEDitor is nothing more than a settings-page with a lot of user-friendly settings for the LED Ring :grin:

But ... no problem. I'll see what the 2.0 mobile app is going to bring us. I just took a quick look at Web API but I may need some startup help, or another app example for that... Maybe I should study the HomeScript source files a little. 😃  
Same (studying) goes voor Promises. For now I copied and adapted some example codes from Athom Developer. It works and I think I understand what's happening... but I may be thinking wrong. I'm a rather easy and fast learner, but hard to convince that I NEED to learn something new... will be my age I guess :grin:
